app/models/manageiq/providers/azure/cloud_manager/provision/cloning.rb

```diff
@@ -1,19 +1,14 @@
 module ManageIQ::Providers::Azure::CloudManager::Provision::Cloning
-  def do_clone_task_check(clone_task_ref)
+  def do_clone_task_check(_clone_task_ref)
     source.with_provider_connection do |azure|
       vms      = ::Azure::Armrest::VirtualMachineService.new(azure)
-      instance = vms.get(clone_task_ref[:vm_name], clone_task_ref[:vm_resource_group])
+      instance = vms.get(dest_name, resource_group.name)
       status   = instance.properties.provisioning_state
       return true if status == "Succeeded"
       return false, status
     end
   end
 
-  def find_destination_in_vmdb(vm_uid_hash)
-    ems_ref = vm_uid_hash.values.join("/")
-    ManageIQ::Providers::Azure::CloudManager::Vm.find_by("lower(ems_ref) = ?", ems_ref.downcase)
-  end
-
```

Comment:
Now that we return a normal ems_ref string this override is unnecessary (also it was missing the :ems_id :scream:)

Here is the core method: https://github.com/ManageIQ/manageiq/blob/4d1900133517c09d90df6a0712aa6ba776a0c9c9/app/models/manageiq/providers/cloud_manager/provision/cloning.rb#L2-L4